api/models.py

```diff
@@ -95,6 +99,11 @@ class User(ModelId):
         description='True if superuser otherwise False'
     )
 
+    @classmethod
+    def create_index(cls, collection):
+        """Create an index to bind unique constraint to username"""
+        collection.create_index("username", unique=True)
```

Comment:
FYI I've double-checked and Mongo DB doesn't recreate an existing index that already exists:
https://www.mongodb.com/docs/manual/reference/method/db.collection.createIndex/#recreating-an-existing-index

This is just to confirm it's OK to just call `create_index()` without first checking whether it already exists.

Reply:
`create_index` will only create an index if it doesn't exist. This explains it: https://stackoverflow.com/questions/61206000/mongodb-index-creation-only-if-not-already-exists

Reply:
Yes that's what I just pointed out from the actual Mongo DB documentation (typically more reliable than forums like SO).

Reply:
Yes, thanks for pointing it out.